### PR TITLE
Creating dev server package

### PR DESCRIPTION
**Pull request by ductiletoaster**
Moving -tasks all around and adding stubbing out the ability to add hooks into the dev server.

### CODE REVIEW

File: packages/pixeloven-webpack-config/src/configs/webpack/client.test.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import * as core from "@pixeloven/core";
+import "jest";
+import sinon, { SinonSandbox } from "sinon";
+import client from "./client";
+
+let sandbox: SinonSandbox;
+
+describe("@pixeloven/webpack", () => {
+    describe("configs", () => {
```
Comment (jimmynotjim) [inline packages/pixeloven-webpack-config/src/configs/webpack/client.test.ts]: If you wanted to reduce the nesting the final output is just a space between each so you could write one describe statement as `@pixeloven/webpack configs webpack`. Also seeing it together makes me realize that the last  `webpack` is probably redundant and unnecessary right?

File: packages/pixeloven-webpack-dev-server/src/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const caller = () => {
+    require("./index");
+};
+
+describe("@pixeloven/webpack-dev-server", () => {
+    describe("index", () => {
+        afterAll(() => {
+            jest.clearAllMocks();
+            jest.restoreAllMocks();
```
Comment (jimmynotjim) [inline packages/pixeloven-webpack-dev-server/src/index.test.ts]: Shouldn't the live above remove any mocks that could possibly be restored?

File: packages/pixeloven-webpack/tslint.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (jimmynotjim) [inline packages/pixeloven-webpack/tslint.json]: Nit: 2 space indentation throughout, not sure how prettier didn't catch this.

**Comment by ductiletoaster** [comment]
Throwing this PR away. Went a slightly different direction

**Pull request closed without merging**